editoast/src/views/train_schedule/simulation_report.rs

```diff
@@ -182,6 +205,74 @@ async fn project_simulation_results(
     })
 }
 
+async fn add_stops_additional_information(
+    stops: Vec<ResultStops>,
+    infra_id: i64,
+    path_waypoints: Vec<PathWaypoint>,
+    db_pool: Data<DbPool>,
+) -> error::Result<Vec<FullResultStops>> {
+    block(move || {
+        let mut conn = db_pool.get()?;
+        let track_ids: Vec<String> = path_waypoints
+            .iter()
+            .map(|pw| pw.location.track_section.0.clone())
+            .collect();
+        let track_sections: Vec<TrackSectionModel> =
+            sql_query(include_str!("sql/track_sections.sql"))
+                .bind::<BigInt, _>(infra_id)
+                .bind::<SqlArray<Text>, _>(&track_ids)
+                .load(&mut conn)?;
+        let track_sections_map: HashMap<String, TrackSectionModel> = HashMap::from_iter(
+            track_sections
+                .iter()
+                .map(|ts| (ts.obj_id.clone(), ts.clone())),
+        );
+        let stops = stops
+            .iter()
+            .zip(path_waypoints.iter())
+            .map(|(s, pw)| {
+                match &track_sections_map
+                    .get(&pw.location.track_section.0)
+                    .unwrap()
+                    .data
+                    .extensions
+                    .sncf
+                {
+                    Some(ext) => FullResultStops {
+                        result_stops: ResultStops {
+                            time: s.time,
+                            position: s.position,
+                            duration: s.duration,
+                        },
+                        id: pw.id.clone(),
+                        name: pw.name.clone(),
+                        line_code: Some(ext.line_code),
+                        track_number: Some(ext.track_number),
+                        line_name: Some(ext.line_name.to_string()),
+                        track_name: Some(ext.track_name.to_string()),
+                    },
+                    None => FullResultStops {
+                        result_stops: ResultStops {
+                            time: s.time,
+                            position: s.position,
+                            duration: s.duration,
+                        },
+                        id: None,
+                        name: None,
+                        line_code: None,
+                        track_number: None,
+                        line_name: None,
+                        track_name: None,
```

Comment:
Could we use a default for this ?
I think we could probably refacto this part of the code to be more simple.

Reply:
we can, `Option` defaults to `None`